File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
var firstUniqChar = function(s) {
    let m = new Map();
    let n = s.length;

    for(const ch of s){
        m.set(ch,(m.get(ch)||0)+1);
    }

    for(let i = 0 ; i<n; i++){
        
        if(m.get(s[i])==1)return i;
    }

    return -1;
};

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        m={}
        for ch in s:
            m[ch]=m.get(ch,0)+1
        
        for i in range(0,len(s)):
            if(m.get(s[i])==1):
                return i
        return -1
        